src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj

```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup Label="Targets required for unit tests to run">
     <TargetFrameworkNetDesktop461>net461</TargetFrameworkNetDesktop461>
-    <TargetFrameworkNetCore>netcoreapp2.1</TargetFrameworkNetCore>
+    <TargetFrameworkNetCore>netcoreapp3.1</TargetFrameworkNetCore>
```

Comment:
Is there a conflict between System.Diagnostics.DiagnosticSource and netcoreapp2.1 or bump to 3.1 is unrelated?

Reply:
Yes, the System.Diagnostics.DiagnosticSource 7.0.2 version needed is only supported for 3.1 onwards.

Reply:
In any case, we should: 

- bump to net fwk 4.6.2
- remove netcoreapp 2.1 (rely on netstandard and net6)
- remove net45

Reply:
Created tasks to track this work

[[Engineering Task] Remove support for netcoreapp 2.1 · Issue #4313 · AzureAD/microsoft-authentication-library-for-dotnet (github.com)](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4313)
[[Engineering Task] Remove support for net45 · Issue #4314 · AzureAD/microsoft-authentication-library-for-dotnet (github.com)](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4314)
[[Engineering Task] Bump net 461 to net 462 · Issue #4315 · AzureAD/microsoft-authentication-library-for-dotnet (github.com)](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4315)